functional/ldap_test.go

```diff
@@ -123,3 +125,52 @@ func TestConnectorLDAPHealthy(t *testing.T) {
 		}
 	}
 }
+
+func TestLDAPPool(t *testing.T) {
```

Comment:
Might pick a more descriptive name. Like "TestMultipleLDAPConns". You aren't testing the pool, you're testing concurrent access :)